.github/workflows/test-e2e.yml

```diff
@@ -51,7 +52,22 @@ jobs:
         # Run only on forks
         if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository }}
 
+      - name: Run the e2e test environment
+        run: mkdir -p test-logs/e2e-test-env && nohup npx ts-node --esm packages/cli/test/scripts/e2e_test_env.ts > test-logs/e2e-test-env/simulation.out 2>&1 &
+      - name: Wait for the node to be ready
+        run: npx wait-port -t 60000 0.0.0.0:5001
+
       - name: E2E tests
         run: yarn test:e2e
         env:
           GOERLI_RPC_URL: ${{ secrets.GOERLI_RPC_URL!=0 && secrets.GOERLI_RPC_URL || env.GOERLI_RPC_DEFAULT_URL }}
+
+      - name: Stop the e2e test environment
+        run: kill -9 $(lsof -t -i:5001)
```

Comment:
Could you move this logic onto a bash script with instructions so it can be recreated locally?

Reply:
Recreating locally have more benefits for everyone. Will update the scripts.